### DISCUSSION TITLE
bot listen from chanel

### DISCUSSION THREAD

**Discussion by yeucodonvn** [telegraf/telegraf#1770] [Q&A]
I don't know how to make the bot wait for and receive a file from a specific channel id.
bot.command and bot.on are two different ids, so it cannot see the file that has been downloaded from the channel.
```
bot.command('upfile', (ctx) => {
    ctx.reply('you can upload');  // ctx.chat.id is channel ID
    bot.on('document', (ctx) => {
            if (ctx.update.message.document) { //ctx.chat.id is bot message id
                ctx.telegram.getFile(ctx.update.message.document.file_id).then((file) => {
                const content = fs.readFileSync('./downloads/' + file.file_path, 'utf8');
                const array = content.split('\n');
                ctx.reply(array);
                });
            }
    });
});
```

**Comment by MKRhere** [comment]
You should **NEVER** nest listeners like this:

```TS
bot.command('upfile', (ctx) => {
    bot.on('document', (ctx) => {
```

All it does is mount a new listener for all documents **every time** the command `/upfile` is triggered. This is terrible because not only does it not do what you wanted, but if this keeps happening, the bot will run out of memory and eventually die.

Consider using scenes.

```TS
// @file: main.js
// you can also use require normally
import { Telegraf, Scene } from "telegraf";
import { document } from "./scenes/document.js";

const stage = new Scenes.Stage([ document ]);
const bot = new Telegraf(token);
bot.use(session()); // Important! Scenes require session first
bot.use(stage.middleware()); // enable our scenes

bot.command("upfile", ctx => {
	// add any conditions you need here
	return ctx.scene.enter("document-collector");
});
```

When a user sends the `/upfile` command, we push them into a scene. The scene code looks like this:

```TS
// @file: scenes/document.js
// you can also use require normally
import { Scene } from "telegraf";
import { message } from "telegraf/filters";

export const document = new Scenes.BaseScene("document-collector");

// send a prompt message when user enters scene
document.enter(ctx => ctx.reply("Upload your document"));

// listen for document messages once user is in the scene
document.on(message("document"), ctx => {
	// ctx.update.message.document;
});

// reply to all other types of messages
document.on("message", ctx => ctx.reply("Invalid message. Upload a document or send /exit"));

// a way to exit the scene
document.command("exit", ctx => ctx.scene.leave());
```

Until you call `ctx.scene.leave()`, the user will be trapped in the scene itself, and you can handle their messages in that context.

See [examples of scenes](https://github.com/feathers-studio/telegraf-docs/tree/4feffcd893a86ed1fcc761279d17328beb4a2f21/examples/scenes). If you need to handle multiple questions in a sequence, you may want to use a Wizard scene. See [examples of wizards](https://github.com/feathers-studio/telegraf-docs/tree/4feffcd893a86ed1fcc761279d17328beb4a2f21/examples/wizards).
